pkgs/development/tools/so/default.nix

```
@@ -0,0 +1,28 @@
+{ stdenv, rustPlatform, fetchFromGitHub, openssl, pkg-config, lib, libiconv, darwin }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "so";
+  version = "0.4.2";
+
+  src = fetchFromGitHub {
+    owner = "samtay";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "09zswxxli9f5ayjwmvqhkp1yv2s4f435dcfp4cyia1zddbrh2zck";
+  };
+
+  cargoSha256 = "1ddbhy1plag4ckbmlyj47wnky7vgmfa68msl3hl25h1lwmzaf1aq";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [
```

Since you have `stdenv` in scope anyway:

```suggestion
  buildInputs = [ openssl ] ++ stdenv.lib.optionals stdenv.isDarwin [
```

Then `lib` can be removed from the argument attrset.